### PR TITLE
feat: implement markdown/html in excerpt

### PR DESCRIPTION
**Pull request by nadalfederer**

### CODE REVIEW

**Comment by nadalfederer** [comment]
@domharrington Hi. Can you review when you are available?
It looks like this now
image

File: packages/api-explorer/src/Doc.jsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ const CodeSample = require('./CodeSample');
 const Response = require('./Response');
 const ResponseSchema = require('./ResponseSchema');
 const EndpointErrorBoundary = require('./EndpointErrorBoundary');
+const markdown = require('../../markdown');
```
Comment (domharrington) [inline packages/api-explorer/src/Doc.jsx]:
```suggestion
const markdown = require('@readme/markdown');
```

File: packages/markdown/__tests__/__snapshots__/index.test.js.snap [hunk 2]
```diff
@@ -1,19 +1,19 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`anchors 1`] = `
-"<p><a href=\\"http://example.com\\" target=\\"_self\\">link</a><br/>
```
Comment (domharrington) [inline packages/markdown/__tests__/__snapshots__/index.test.js.snap]: Why doesn't this add </br>'s to the end of the lines anymore?

File: packages/markdown/__tests__/__snapshots__/index.test.js.snap [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
+<a href=\\"/blog/slug\\" target=\\"_self\\">blog</a>
+<a href=\\"/page/slug\\" target=\\"_self\\">page</a>
 </p>"
 `;
 
 exports[`check list items 1`] = `
 "<ul>
-<li><input type=\\"checkbox\\" disabled=\\"\\"/> checklistitem1</li>
-<li><input type=\\"checkbox\\" checked=\\"\\" disabled=\\"\\"/> checklistitem1</li>
+<li class=\\"task-list-item\\"><input type=\\"checkbox\\" disabled=\\"\\"/> checklistitem1</li>
```
Comment (domharrington) [inline packages/markdown/__tests__/__snapshots__/index.test.js.snap]: How come this class is added now?

File: packages/markdown/__tests__/__snapshots__/index.test.js.snap [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
 
 exports[`anchors 1`] = `
-"<p><a href=\\"http://example.com\\" target=\\"_self\\">link</a><br/>
-<a href=\\"\\" target=\\"_self\\">xss</a><br/>
-<a href=\\"/docs/slug\\" target=\\"_self\\" class=\\"doc-link\\" data-sidebar=\\"slug\\">doc</a><br/>
-<a href=\\"/reference#slug\\" target=\\"_self\\">ref</a><br/>
-<a href=\\"/blog/slug\\" target=\\"_self\\">blog</a><br/>
-<a href=\\"/page/slug\\" target=\\"_self\\">page</a><br/>
+"<p><a href=\\"http://example.com\\" target=\\"_self\\">link</a>
+<a href=\\"javascript:alert\\" target=\\"_self\\">xss</a>
```
Comment (domharrington) [inline packages/markdown/__tests__/__snapshots__/index.test.js.snap]: Is this safe?

File: packages/markdown/__tests__/__snapshots__/index.test.js.snap [hunk 5]
```diff
@@ -25,13 +25,13 @@ exports[`code samples 1`] = `
 `;
 
 exports[`emojis 1`] = `
-"<p><img src=\\"/img/emojis/joy.png\\" alt=\\":joy:\\" title=\\":joy:\\" class=\\"emoji\\" align=\\"absmiddle\\" height=\\"20\\" width=\\"20\\"/><br/>
```
Comment (domharrington) [inline packages/markdown/__tests__/__snapshots__/index.test.js.snap]: Doesn't this mean our emoji parsing has broken?

File: packages/markdown/__tests__/__snapshots__/index.test.js.snap [hunk 6]
```diff
@@ -55,10 +55,32 @@ exports[`list items 1`] = `
 exports[`should strip out inputs 1`] = `""`;
 
 exports[`tables 1`] = `
-"<div class=\\"marked-table\\"><table>
-<thead><tr><th>Tables</th><th style=\\"text-align:center\\">Are</th><th style=\\"text-align:right\\">Cool</th></tr></thead>
-<tbody><tr><td>col 3 is</td><td style=\\"text-align:center\\">right-aligned</td><td style=\\"text-align:right\\">$1600</td></tr><tr><td>col 2 is</td><td style=\\"text-align:center\\">centered</td><td style=\\"text-align:right\\">$12</td></tr><tr><td>zebra stripes</td><td style=\\"text-align:center\\">are neat</td><td style=\\"text-align:right\\">$1</td></tr></tbody>
-</table></div>"
+"
```
Comment (domharrington) [inline packages/markdown/__tests__/__snapshots__/index.test.js.snap]: How come there's now so much white space?

File: packages/markdown/__tests__/index.test.js [hunk 7]
```diff
@@ -96,13 +96,12 @@ test('should render nothing if nothing passed in', () => {
   expect(markdown('')).toBe(null);
 });
 
-test('`correctnewlines` option', () => {
```
Comment (domharrington) [inline packages/markdown/__tests__/index.test.js]: We still need this option to work. Is it incompatible with using rehype directly?

File: packages/markdown/__tests__/index.test.js [hunk 8]
```diff
@@ -113,29 +112,34 @@ test('glossary', () => {
...
+});
 
-  test('should escape unknown tags', () => {
-    expect(markdown('<unknown-tag>Test</unknown-tag>')).toBe(
-      '<p>&lt;unknown-tag&gt;Test&lt;/unknown-tag&gt;</p>\n',
-    );
-  });
+test('should allow certain attributes', () => {
+  expect(shallow(markdown('<p id="test">Test</p>')).html()).toBe(
+    '<p id="user-content-test">Test</p>',
```
Comment (domharrington) [inline packages/markdown/__tests__/index.test.js]: Where does this id come from?

File: packages/markdown/__tests__/index.test.js [hunk 9]
```diff
@@ -113,29 +112,34 @@ test('glossary', () => {
...
+  );
+});
 
-  test('should allow certain attributes', () => {
```
Comment (domharrington) [inline packages/markdown/__tests__/index.test.js]: Why did you need to remove this test? Can you add back in to make sure the changes you've made are compliant with the old functionality?

File: packages/markdown/index.js [hunk 10]
```diff
@@ -1,9 +1,13 @@
 const React = require('react');
 const remark = require('remark');
-const reactRenderer = require('remark-react');
-const breaks = require('remark-breaks');
```
Comment (domharrington) [inline packages/markdown/index.js]: I think we still need this for the correctnewlines option

**Comment by nadalfederer** [comment]
@domharrington Hi. I've fixed current bugs

File: packages/markdown/__tests__/__snapshots__/index.test.js.snap [hunk 11]
```diff
@@ -56,8 +56,30 @@ exports[`should strip out inputs 1`] = `""`;
 
 exports[`tables 1`] = `
 "<div class=\\"marked-table\\"><table>
-<thead><tr><th>Tables</th><th style=\\"text-align:center\\">Are</th><th style=\\"text-align:right\\">Cool</th></tr></thead>
-<tbody><tr><td>col 3 is</td><td style=\\"text-align:center\\">right-aligned</td><td style=\\"text-align:right\\">$1600</td></tr><tr><td>col 2 is</td><td style=\\"text-align:center\\">centered</td><td style=\\"text-align:right\\">$12</td></tr><tr><td>zebra stripes</td><td style=\\"text-align:center\\">are neat</td><td style=\\"text-align:right\\">$1</td></tr></tbody>
+<thead>
+<tr>
+<th>Tables</th>
+<th align=\\"center\\">Are</th>
```
Comment (domharrington) [inline packages/markdown/__tests__/__snapshots__/index.test.js.snap]: What is it that adds these attributes instead of `text-align: center`? These are deprecated: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/td#attr-align

**Comment by nadalfederer** [comment]
@domharrington Hi. I've fixed it. As I investigate that there is a bug in a library, I've found that it just got fixed recently.

**Comment by domharrington** [comment]
Thanks! I refactored it slightly to simplify and ensure we only parse the content once: https://github.com/readmeio/api-explorer/blob/c053ee6ef0bc434eb199402d10bab16ee4926ea0/packages/markdown/index.js#L47

But good stuff!
